awscrt/http.py

```diff
@@ -112,6 +112,29 @@ def new(cls,
             If successful, the Future will contain a new :class:`HttpClientConnection`.
             Otherwise, it will contain an exception.
         """
+        return HttpClientConnection._generic_new(
+            cls,
+            host_name,
+            port,
+            bootstrap,
+            socket_options,
+            tls_connection_options,
+            proxy_options)
+
+    @classmethod
+    def _generic_new(
+            _cls,
+            expected_cls,
```

Comment:
a `classmethod` is just a `staticmethod` that automatically passes `cls`
if you don't want automatic `cls`, do `staticmethod` instead
```suggestion
    @staticmethod
    def _generic_new(
            expected_cls,
```